src/test/tv/codely/scala_intro_examples/lesson_09_oop/exercise_solutions/joan_miralles_solution/EmailTest.scala

```diff
@@ -0,0 +1,46 @@
+package tv.codely.scala_intro_examples.lesson_09_oop.exercise_solutions.joan_miralles_solution
+
+import java.security.InvalidParameterException
+
+import org.scalatest.{Matchers, WordSpec}
+
+class EmailTest extends WordSpec with Matchers {
+
+  private val validEmail = "some@email.com"
+  private val anotherValidEmail = "a@e.t"
+  private val invalidEmail = "some.email.com"
+  private val anotherInvalidEmail = "some@email"
+
+  "Email Class" should {
+    "provide an apply method in the companion object in order to construct new Email from complete string" in {
+      "Email(completeEmail = validEmail)" should compile
+    }
+
+    "return a new Email instance given a complete email" in {
+      val email = Email(validEmail)
+      assert(email.isInstanceOf[Email])
```

Comment:
Redundant assertion. Same reason as in the previous comment. If we couldn't instantiate an `Email` in the previous statement `val email = Email(validEmail)`, it will throw an exception making our test fail.